### ISSUE TITLE
How to center div box?

### ISSUE THREAD

**Issue by quantum-satire** [ebeshero/DHClass-Hub#348]
I'm getting around to coding my resumé for our Newtfire site and I'm trying to enclose all of my text on the page in a boundary div box to make it look more condensed on the page. Here is what I have right now:
![screen shot 2017-09-24 at 11 30 08 am](https://user-images.githubusercontent.com/28074589/30784076-fe20d90c-a11b-11e7-9b13-c66fc744cc0b.png)
I thought that by setting the `margin` to `auto` the box would align itself to the center of the webpage. Am I doing something wrong?

**Comment by gabikeane** [comment]
Try pushing your code to the sandbox so someone else can take a peek!  While you're doing that, check to make sure that you've used the class on the div the right way, you've saved and reloaded your documents, and that you've refreshed your browser.  I don't know whether you discussed this in the Greensburg class, but you can preview your site using Google Chrome so you don't have to bother with reuploading to the server every time (just go into your file explorer, right click, and select 'Open with'.

**Comment by ebeshero** [comment]
I think we mentioned this with keystrokes: You can go to your browser, and open a local file with: 
`control + o` (the letter o) on a PC or
`command + o` on a Mac. 
Refreshing the CSS might be a bit laggy--you need to hit refresh your browser, sometimes a couple of times (sigh).

**Comment by quantum-satire** [comment]
@gabikeane [Here](https://github.com/ebeshero/DHClass-Hub/commit/fda72cec713b5d80112a516142db625ce53ef801) is a link! I did make sure to close the Chrome tab and open another each time I made a change to my code.

**Comment by ebeshero** [comment]
@quantum-satire @gabikeane We're also encouraging people to work **locally** in their own personal repos on these assignments--so instead of having to push to the DHClass-Hub sandbox, prepare a web directory in your personal GitHub repo, and use that as your local directory for sending files to newtfire, as well. That way you can track your own files on your websites as you develop them. And that's how we'll be working on the project websites too, as you have your project repos and your project web directories.

**Comment by gabikeane** [comment]
ctrl + r or command + r for a hard reload btw! that often helps!

**Comment by ebeshero** [comment]
Sorry for any protocol confusion here--this is likely a new way of working for the Pittsburgh group. But in practice, so we **don't have to duplicate** saving things in multiple repos, when you're working on your personal website, work in your personal GitHub repo, and when you want help, go to the DHClass-Hub Issues board and direct us to look in on your personal repo. 

The same will hold as you're working in your project repos. You can push code here, too, but we want to minimize having to duplicate your files in lots of separate GitHub locations--that gets messy after a while.

**Comment by quantum-satire** [comment]
@ebeshero I'll delete the file from the Sandbox and upload it to my repo!

**Comment by ebeshero** [comment]
@quantum-satire Thanks--I'm trying to get people to use those personal repos! :+1:  Also, I'm digging into your code and am a little mystified (as you are) about why that margin setting isn't working. Something else must be interfering. Also, I'm on my LInux machine--browsers are often wonky on this thing, so I'll switch over to my Mac to take a look... 

@gabikeane In case you're mystified by Ally's inline CSS, she's promised to move it into a separate CSS file--this is just her preferred method of drafting her code for the first run.

**Comment by ebeshero** [comment]
@quantum-satire Send a link here to your personal repo when you've got it pushed there.

**Comment by quantum-satire** [comment]
@ebeshero [Here](https://github.com/quantum-satire/DHClass-notes/blob/master/website/resume.html) is the new link! I am working from my Mac now, I have yet to test it on my Linux now that you've mentioned it.

**Comment by jonhoranic** [comment]
@quantum-satire Jumping in because I heard the words "CSS" and "help" coming through the æther. Due to my slight "rivalry" with CSS I can almost hear every web-dev's silent internal screams when a <div> element is just ever so slightly overlapping other content...although a few of those might also be some of my own, it is hard to tell.

Nevertheless, I can share a little bit of wisdom from my time experimenting with CSS. Centering is not as simple as it may seem, and it takes quite a lot of set up if you do it solely by hand. Recently I have been experimenting with flex-boxes, which are pretty much just element containers that act as "backboards" that your nested items will be "posted on" and both the flex-box and its child items (individually or as a group) can be styled separately to achieve a variety of interesting effects.

Here is a [helpful tutorial link](https://internetingishard.com/html-and-css/flexbox/) from last year that really helped me to discover better webpage centering and positioning. See what results this brings and be sure to tell CSS that I said hello!

**Comment by quantum-satire** [comment]
@jonhoranic I'll be sure to read through this--thank you for the tutorial!

**Comment by gabikeane** [comment]
Just to avoid confusion and keep things where others can find them, I'm going to comment on your code here.  
You're running into problems because you are attempting to center your box div within a left-aligned container div.  Fix this by setting the margin to auto on the container div, or by disposing of the container div completely and only using and styling the box div.

I want to note here that the best way to troubleshoot CSS styling is by isolating the code that gives you a problem, sometimes by commenting things out or by creating a whole other troubleshoot document.  In this case I realized where your issue was as soon as I went to isolate the problem code, but when I looked at it as a whole I couldn't see what was messing with you.
Let us know if this helps!
